src/actions/sources.js

```diff
@@ -101,8 +101,15 @@ export function newSources(sources: Source[]) {
       source => !getSource(getState(), source.id)
     );
 
+    dispatch({
+      type: "ADD_SOURCES",
+      sources
+    });
+
     for (const source of filteredSources) {
-      await dispatch(newSource(source));
+      await dispatch(loadSourceMap(source));
```

Comment:
Yep. I tried to break this work up a bit so that we can land it piecemeal. This logic will be touched in another PR